src/coreclr/nativeaot/Runtime/FinalizerHelpers.cpp

```diff
@@ -46,9 +46,6 @@ uint32_t WINAPI FinalizerStart(void* pContext)
 
     g_pFinalizerThread = PTR_Thread(pThread);
 
-    // We have some time until the first finalization request - use the time to calibrate normalized waits.
-    EnsureYieldProcessorNormalizedInitialized();
```

Comment:
How is the measurement going to be triggered when this is deleted?

Reply:
I'm still trying to figure this out, I'm not very familiar with Native AOT in general so I'd appreciate any suggestions

Reply:
It looks like we would need to call `YieldProcessorNormalization::PerformMeasurement()` from here or add a `EnsureYieldProcessorNormalizedInitialized()` entry point to the new code that simply calls `YieldProcessorNormalization::PerformMeasurement()`

Reply:
Do you happen to know if this function is called every ~4 seconds or faster than that? Currently we let `YieldProcessorNormalization::PerformMeasurement()` run every ~4 s so if that's the case, I believe we may add here the same call as in CoreCLR
```cpp
if (YieldProcessorNormalization::IsMeasurementScheduled())
    {
        GCX_PREEMP();
        YieldProcessorNormalization::PerformMeasurement();
    }
```

Reply:
`FinalizerStart` function is called once per process. It is equivalent of `FinalizerThreadStart` function in regular CoreCLR.

I think you want to follow the same structure as in regular CoreCLR: Trigger the measurement from ScheduleMeasurementIfNecessary by calling `RhEnableFinalization` (it is equivalent of `FinalizerThread::EnableFinalization` in regular CoreCLR) and then add the measurement to loop in `ProcessFinalizers()`.

Reply:
> Do you happen to know if this function is called every ~4 seconds or faster than that?

I am not sure. The whole deal with measuring duration of something that is proportional to clock cycle is not very precise, since the clock cycle can change drastically and many times per second and will be different for every core. Unless machine is configured into HighPerformance power plan, every measurement is a bit of a coin toss and will produce the same result with the same error margins. 

The main purpose of calibration is to continue using historically hard-coded spin counts in numerous places where we spinwait while allowing that to work on systems with vastly different pause durations (i.e. on post-skylake intel CPUs pause takes ~140 cycles, pre-skylake is about ~10 cycles). For such purpose the callibration is precise enough.

I am not sure about the value of redoing this over and over. 
Perhaps to support scenarios where a VM is migrated between pre/post skylake machines.

Reply:
I guess we can add a periodic call PerformMeasurement in NativeAOT and see what happens. 

My guess - nothing will change, just a bit more time spent in PerformMeasurement.

Reply:
There is value in having the same behavior though.  
If we remeasuring (or the whole calibration deal) could be somehow avoided or improved, it would make sense to do for both runtimes.

Reply:
IIRC there's a good reason to keep re-doing measurements, so probably keeping this behaviour in Native AOT would be better, I believe @kouvel or @mangod9 may elaborate better

Reply:
The measurements done are very short and can be perturbed by CPU activity, the rolling min helps to stabilize it over time.